### ISSUE TITLE
Financial functions ipmt and ppmt result in incorrect values for per=1

### ISSUE THREAD

**Issue by pverleg** [numpy/numpy#9495]
The numpy.ipmt and numpy.ppmt functions seem to misbehave when per=1. The ipmt+ppmt still equals the result of numpy.pmt, but as you can see below, ipmt results in 0 for per=1, while ppmt equals the full amount of principal+interest. 

I noticed that https://github.com/numpy/numpy/blob/master/numpy/lib/financial.py contains an exception for per=1, but I do not think that should be there:
`ipmt = np.where(np.logical_and(when == 1, per == 1), 0.0, ipmt)`

```
# Principal calculation (notice second line)
numpy.ppmt(rate=0.001988079518355057, per=0, nper=360, pv=300000, fv=0, when="begin")=-568.922801885
numpy.ppmt(rate=0.001988079518355057, per=1, nper=360, pv=300000, fv=0, when="begin")=-1165.29433577
numpy.ppmt(rate=0.001988079518355057, per=2, nper=360, pv=300000, fv=0, when="begin")=-571.18717807
numpy.ppmt(rate=0.001988079518355057, per=3, nper=360, pv=300000, fv=0, when="begin")=-572.3227436
# ...etc

# Interest calculation (notice second line)
numpy.ipmt(rate=0.001988079518355057, per=0, nper=360, pv=300000, fv=0, when="begin")=-596.37153388933
numpy.ipmt(rate=0.001988079518355057, per=1, nper=360, pv=300000, fv=0, when="begin")=0.0
numpy.ipmt(rate=0.001988079518355057, per=2, nper=360, pv=300000, fv=0, when="begin")=-594.1071577047084
numpy.ipmt(rate=0.001988079518355057, per=3, nper=360, pv=300000, fv=0, when="begin")=-592.9715921748406
# ...etc
```

**Comment by eric-wieser** [comment]
Do we actually have any contributors experienced with `np.lib.financial`? I'm beginning to think that:

* We need separate labels for `np.lib.*`
* Kicking out `np.lib.financial` into a separate library is becoming more appealing

**Comment by eric-wieser** [comment]
Seems this function has always been this way, and was introduced in 015cada57ddd00d866e2a8ef396f83582c192884, by an anonymous author

**Comment by shoyer** [comment]
@eric-wieser  I agree, these concerns are worth raising on the mailing list. If we don't have anyone to maintain it, then keeping this module around is not doing anyone a service.

I have no idea why Travis thought it was a good idea to write this module back in 2008 :).

**Comment by eric-wieser** [comment]
I don't think making a label warrants a mailing list post - I've gone ahead and relabelled all the np.lib.financial issues and PRs that I can find, to get a feel for how often people actually ask about it.

I might get around to asking the mailing list about generally maintaining it though. Related to what to do in general with `financial` is #2880
